ext/strscan/strscan.c

```diff
@@ -714,14 +706,24 @@ strscan_do_scan(VALUE self, VALUE pattern, int succptr, int getstr, int headonly
         }
     }
     else {
+        StringValue(pattern);
         rb_enc_check(p->str, pattern);
         if (S_RESTLEN(p) < RSTRING_LEN(pattern)) {
             return Qnil;
         }
-        if (memcmp(CURPTR(p), RSTRING_PTR(pattern), RSTRING_LEN(pattern)) != 0) {
-            return Qnil;
+
+        if (headonly) {
+            if (memcmp(CURPTR(p), RSTRING_PTR(pattern), RSTRING_LEN(pattern)) != 0) {
+                return Qnil;
+            }
+            set_registers(p, RSTRING_LEN(pattern));
+        } else {
+            long pos = rb_memsearch(RSTRING_PTR(pattern), RSTRING_LEN(pattern), CURPTR(p), S_RESTLEN(p), rb_enc_get(pattern));
```

Comment:
Could you fold a long line?

```suggestion
            long pos = rb_memsearch(RSTRING_PTR(pattern), RSTRING_LEN(pattern),
                                    CURPTR(p), S_RESTLEN(p), rb_enc_get(pattern));
```

BTW, `rb_memsearch()` document https://github.com/ruby/ruby/blob/ddbd64400199fd408d23c85f9fb0d7f742ecf9e1/include/ruby/internal/encoding/string.h#L307-L319 is wrong.
(`x` not `y` is the query string.)

Could you report it to upstream?

Reply:
> Could you fold a long line?

OK.
I see.

> BTW, `rb_memsearch()` document https://github.com/ruby/ruby/blob/ddbd64400199fd408d23c85f9fb0d7f742ecf9e1/include/ruby/internal/encoding/string.h#L307-L319 is wrong. (`x` not `y` is the query string.)
> 
> Could you report it to upstream?

OK!